### PR TITLE
Catch up on tech dept on europe

### PR DESCRIPTION
**Pull request by SondreNjaastad**
- Location now uses the correct endpoint
- rename vehicle files to make it the same across the project
- Fix VehicleStatus interface, rename and add locked property
- Moved constants to constants/europe.ts
- Updated tests accoring to changes above

### CODE REVIEW

File: lib/controllers/european.controller.ts [hunk 1]
```diff
@@ -75,16 +84,18 @@ export class EuropeanController extends SessionController {
           password: this.userConfig.password,
         },
         cookieJar,
+      }).catch(err => {
+        logger.debug(`Login failed: ${err}`);
```
Comment (Hacksore) [inline lib/controllers/european.controller.ts:88]: This won't expand the error and will likely output `[object Object]` cause I have not fixed the Winston logger yet.

File: lib/controllers/european.controller.ts [hunk 2]
```diff
@@ -75,16 +84,18 @@ export class EuropeanController extends SessionController {
...
-      const queryString = authCodeResponse.body.redirectUrl.split('?')[1];
-      const urlParams = new URLSearchParams(queryString);
-      const authToken = urlParams.get('code');
-
-      if (authToken !== null) {
-        this.session.refreshToken = authToken;
-      } else {
-        throw new Error('@EuropeControllerLogin: AuthCode was not found');
+      if(authCodeResponse){
+        const regexMatch = /.*code=(\w*)&.*/g.exec(authCodeResponse.body.redirectUrl);
```
Comment (Hacksore) [inline lib/controllers/european.controller.ts:93]: Does this regex work with codes that have dashes? I ran into an issue that codes were coming back with dashes and getting failures.

File: lib/index.ts [hunk 3]
```diff
@@ -71,7 +71,7 @@ class BlueLinky extends EventEmitter {
       this.emit('ready', this.vehicles);
       return response;
     } catch (error) {
-      throw new Error(error.message);
+      return error;
```
Comment (Hacksore) [inline lib/index.ts:74]: What's the thought behind returning an error rather than throwing it?

File: lib/vehicles/american.vehicle.ts [hunk 4]
```diff
@@ -208,7 +209,7 @@ export default class AmericanVehicle extends Vehicle {
         charging: vehicleStatus?.evStatus?.batteryCharge,
         batteryCharge: vehicleStatus?.battery?.batSoc,
       },
-    };
+    } as VehicleStatus;
```
Comment (Hacksore) [inline lib/vehicles/american.vehicle.ts:212]: I think this is redundant somewhat as the return type of the method is a `VehicleStatus`. Nevertheless, it doesn't hurt anything keeping it in.

File: lib/controllers/european.controller.ts (hunk 2 above)
Comment (SondreNjaastad) [reply lib/controllers/european.controller.ts:93]: Have yet to see a dash in the codes, but we have to deal with that then

File: lib/index.ts (hunk 3 above)
Comment (SondreNjaastad) [reply lib/index.ts:74]: Not having unhandled rejects in promises, as it provides no log info

File: lib/vehicles/american.vehicle.ts (hunk 4 above)
Comment (SondreNjaastad) [reply lib/vehicles/american.vehicle.ts:212]: While developing its needed to hightlight any missing properties etc

**Pull request merged**